pegasus/sites.v3/code.org/public/administrators.erb

```diff
@@ -257,6 +257,8 @@ Did you know the best way to encourage students to try CS is to tell them they
 
 <p style="font-size:11px">“Human mind” icon made by <a href="https://smashicons.com/" title="Smashicons" target="_blank">Smashicons</a>, “Developer” icon made by <a href="https://www.flaticon.com/free-icon/developer_1508765?term=woman%20computer&page=2&position=60" title="mynamepong" target="_blank">mynamepong</a>, “Like” icon made by <a href="https://www.flaticon.com/authors/good-ware" title="Good Ware" target="_blank">Good Ware</a>. All icons sourced from <a href="https://www.flaticon.com/" title="Flaticon" target="_blank"> www.flaticon.com</a>.</p>
 
+<%= view :analytics_event_log_helper, event_name: "ADMIN_PAGE_VISITED_EVENT" %>
```

Comment:
passing the js key by name looks a bit brittle because we won't really know if an invalid key name gets passed in. Instead it would be better to pass the actual event name. To do this, I would recommend one of two approaches:
1. if the event types you are adding are all server-side only, you could move the hashmap defining these events into ruby code
2. if the events are shared between server and client, or if you want them to be all defined in one place, you could  follow the approach used by `shared_constants.rb` and `sharedConstants.js` to share these constants between ruby and js, and then use the ruby version of them here. if you choose this option, you should probably define a new pair of files named something like `shared_amplitude_constants.rb` and `sharedAmplitudeConstants.rb`.

Reply:
After discussion with Dave, I've opted for choice 1 so the event constants will be split up into different files in the pegasus and apps directories.